### PR TITLE
Improve accuracy of retransmit counting on kernels >= 4.7

### PR DESCRIPTION
**Pull request by brycekahle**
### What does this PR do?

In kernel 4.7 a [change](https://github.com/torvalds/linux/commit/10d3be569243def8d92ac3722395ef5a59c504e6) was made to send multiple retransmits in the same socket buffer. In order to accurately track retransmitted segments, we need to use the newly added 3rd argument, named `segs` (for segments).

### Motivation

Improved accuracy of the reported retransmits number.

### Additional Notes

This installs a different probe depending on whether the kernel version is >= 4.7 or not. This is the same pattern we use for some of the < 4.1 probes.

The runtime compilation version doesn't need this switch, so it doesn't have a "pre 4.7" probe at all. This means we have to only declare that probe when using the prebuilt tracer. This code handles that logic.

### Describe your test plan

CI Kitchen tests are passing. I'm not sure there is a way to create the situation in CI that would result in `segs > 1`.

### CODE REVIEW

**Review by heyronhay — APPROVED** [review]
One comment request, looks good otherwise!

**Review by leeavital — COMMENTED** [review]
Do we need to consider the 3.x kernel version for centos  ?

**Comment by brycekahle** [comment]
> Do we need to consider the 3.x kernel version for centos ?

This only changes the behavior on kernels >= 4.7

File: pkg/network/config/config_linux.go [hunk 1]
```diff
@@ -20,11 +27,16 @@ func (c *Config) EnabledProbes(pre410Kernel bool) (map[probes.ProbeName]struct{}
 		enabled[probes.TCPCleanupRBuf] = struct{}{}
 		enabled[probes.TCPClose] = struct{}{}
 		enabled[probes.TCPCloseReturn] = struct{}{}
-		enabled[probes.TCPRetransmit] = struct{}{}
 		enabled[probes.InetCskAcceptReturn] = struct{}{}
 		enabled[probes.TCPv4DestroySock] = struct{}{}
 		enabled[probes.TCPSetState] = struct{}{}
 
+		if !runtimeTracer && kv < kernel.VersionCode(4, 7, 0) {
+			enabled[probes.TCPRetransmitPre470] = struct{}{}
```
[the remark this answers is not in the document]
Comment (brycekahle) [reply pkg/network/config/config_linux.go:35]: Done!

**Comment by leeavital** [comment]
RHEL maintains its own versioning scheme, where the kernel versions we get show as 3.x. However, lots of features are backported.   It looks like centos8 has the `tcp_retransmit_skb` signature where segments is in the third parameters.  
 https://github.com/kernelim/linux/blob/centos8/include/net/tcp.h#L576

**Comment by brycekahle** [comment]
> It looks like centos8 has the `tcp_retransmit_skb` signature where segments is in the third parameters

CentOS 8 has kernel version 4.18, so this code would correctly handle it. It looks like the relevant commit didn't land until CentOS 8 either: https://github.com/kernelim/linux/commit/10d3be569243def8d92ac3722395ef5a59c504e6

**Review by leeavital — APPROVED** [review]

**Pull request merged**